limbus/core/params.py

```diff
@@ -1,40 +1,26 @@
 """Classes to define set of parameters."""
 from __future__ import annotations
-from typing import Any, Iterator, Iterable
+from typing import Any, Iterator, Iterable, Callable
+from abc import ABC, abstractmethod
 
 # Note that Component class cannot be imported to avoid circular dependencies.
 # Since it is only used for type hints we import the module and use "component.Component" for typing.
 from limbus.core import component
-from limbus.core.param import Param, NoValue, InputParam, OutputParam
+from limbus.core.param import Param, NoValue, InputParam, OutputParam, PropParam
 
 
-class Params(Iterable):
+class Params(Iterable, ABC):
     """Class to store parameters."""
 
     def __init__(self, parent_component: None | "component.Component" = None):
         super().__init__()
         self._parent = parent_component
 
-    def declare(self, name: str, tp: Any = Any, value: Any = NoValue(), arg: None | str = None) -> None:
-        """Add or modify a param.
-
-        Args:
-            name: name of the parameter.
-            tp: type (e.g. str, int, list, str | int,...). Default: typing.Any
-            value (optional): value for the parameter. Default: NoValue().
-            arg (optional): Component argument directly related with the value of the parameter. Default: None.
-                            E.g. this is useful to propagate datatypes and values from a pin with a default value to
-                            an argument in a Component (GUI).
-
-        """
-        if isinstance(value, Param):
-            value = value.value
-        setattr(self, name, Param(name, tp, value, arg, self._parent))
-
-    def __getattr__(self, name: str) -> Param:  # type: ignore  # it should return a Param
-        """Trick to avoid mypy issues with dinamyc attributes."""
-        ...
 
+    @abstractmethod
+    def declare(self, *args, **kwargs) -> None:
+        """Add or modify a param."""
+        raise NotImplementedError
```

Comment:
```suggestion
        raise NotImplementedError

```

Reply:
I do not understand teh suggested change... it is the same code as it is XD

Reply:
end of line was empty i believe